hazelcast/src/main/java/com/hazelcast/jet/impl/operation/InitExecutionOperation.java

```diff
@@ -90,16 +94,59 @@ protected CompletableFuture<?> doRun() {
                 caller);
 
         ExecutionPlan plan = deserializePlan(serializedPlan);
-        if (isLightJob) {
-            return service.getJobExecutionService().runLightJob(jobId(), executionId, caller,
-                    coordinatorMemberListVersion, participants, plan);
+        InitExecutionOffload initExecutionAction = new InitExecutionOffload(service, plan);
+        boolean isCooperative = plan.getVertices().stream()
+                .allMatch(v -> {
+                    ProcessorSupplier supplier = v.processorSupplier();
+                    if (!supplier.initIsCooperative()) return false;
+                    return supplier.get(1).iterator().next().initIsCooperative(); // todo: better detection?
```

Comment:
I'm not sure we can do this. We can't ask the `ProcessorSupplier` to provide 1 processor, if local parallelism is different. Some processors split the work among expected number of processors. They also can track processors they created. This is not a good idea...